### PR TITLE
implemented derived quantities

### PR DESCRIPTION
**Pull request by MauriceHendrix**
## Description

Added code to deal with derived quantities

## Motivation and Context

Symbols can be annotated as derived quantities or with expose_annotated_variables option variables that have metadata and are derived quantities are include in chaste code

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated all documentation necessary.
- [ ] I have checked spelling in (new) comments.

##Testing
- [X] Testing is done automaticly and codecov shows test coverage
- [ ] This cannot be tested automatically

### CODE REVIEW

File: weblab_cg/_generator_chaste_cg.py [hunk 1]
```diff
@@ -138,6 +143,20 @@ def __init__(self, model, class_name, file_name):
...
+    @property
+    def expose_annotated_variables(self):
+        return self._expose_annotated_variables
+
+    @expose_annotated_variables.setter
+    def expose_annotated_variables(self, value):
+        self._expose_annotated_variables = value
+        self._derived_quant = self._get_derived_quant()
+        self._formatted_derived_quant = self._format_derived_quant()
+        self._derived_qunat_eqs = self._get_derived_qunat_eqs()
```
Comment (jonc125) [inline weblab_cg/_generator_chaste_cg.py]: Check your spelling!

File: weblab_cg/_generator_chaste_cg.py [hunk 2]
```diff
@@ -353,7 +372,36 @@ def _get_derivative_equations(self):
...
+        """ Get all derived quantities
+
+            Note: the result depends on self._expose_annotated_variables to determine whether or not to include
+            variables exposed with oxford metadata that are derived quantities but are not annotated as such"""
+        if self._expose_annotated_variables:
+            # Combined and sorted in document order
+            return \
+                sorted(self._derived_quant_annotated + self._get_derived_quant_exposed(), key=lambda v: v.order_added)
+        else:
+            return self._derived_quant_annotated
```
Comment (jonc125) [inline weblab_cg/_generator_chaste_cg.py]: Are these already sorted? Probably worth a comment to that effect.

File: weblab_cg/tests/test_chaste_cg_develop.py [hunk 3]
```diff
@@ -162,9 +162,6 @@ def is_same(eq1, eq2):
         eq1 = sympy.sympify(eq1)
         eq2 = sympy.sympify(eq2)
 
-        if eq1.free_symbols != eq2.free_symbols:
```
Comment (jonc125) [inline weblab_cg/tests/test_chaste_cg_develop.py]: Why remove this?

Comment (MauriceHendrix) [reply weblab_cg/tests/test_chaste_cg_develop.py]: becuase it was telling me that 1.0 and 1.0 + 0.0*X wasn't the same which it is!

File: weblab_cg/_generator_chaste_cg.py (hunk 1 above)
Comment (MauriceHendrix) [reply weblab_cg/_generator_chaste_cg.py]: looks good to me! (quant is a deliberate shortening)

Comment (jonc125) [reply weblab_cg/_generator_chaste_cg.py]: qunat != quant :)

Comment (MauriceHendrix) [reply weblab_cg/_generator_chaste_cg.py]: o now I see it

**Comment by MauriceHendrix** [comment]
> This one looks like a mistake!



> For the future, is it worth thinking about excluding changes like this from commits? E.g. https://stackoverflow.com/questions/40794149/how-do-i-tell-gitignore-or-git-itself-to-ignore-lines-containing-todo?noredirect=1&lq=1

Actually it's not about that it's that the dev test is just a tool for myself, so I don't want it to run on ci which it doesn't) but also I don't want to check-in every change to reference files I'm working with. 
These are mostly pycml generated files (so no #todo)

**Comment by MauriceHendrix** [comment]
> This one looks like a mistake!

yes I was trying to get rid of double blank lines, very fidly

**Pull request merged**